arviz/plots/ppplot.py

```diff
@@ -0,0 +1,182 @@
+    prior_kwargs : dicts, optional
+        Additional keywords passed to `arviz.plot_kde` for prior/predictive groups.
+    posterior_kwargs : dicts, optional
+        Additional keywords passed to `arviz.plot_kde` for posterior/predictive groups.
+    observed_kwargs : dicts, optional
+        Additional keywords passed to `arviz.plot_kde` for observed_data group.
+    backend: str, optional
+        Select plotting backend {"matplotlib","bokeh"}. Default "matplotlib".
+    backend_kwargs: bool, optional
+        These are kwargs specific to the backend being used. For additional documentation
+        check the plotting method of the backend.
+    show : bool, optional
+        Call backend show function.
+
+    Returns
+    -------
+    axes : a numpy 2d array of matplotlib axes. Returned object will have shape (nvars, 3) where the last column is the combined plot and the first columns are the single plots.
+
+    Examples
+    --------
+    Plot the prior/posterior plot for specified vars and coords.
+
+    .. plot::
+        :context: close-figs
+
+        >>> import arviz as az
+        >>> data = az.load_arviz_data('radon')
+        >>> az.plot_pp(data, var_names=["defs"], coords={"team" : ["Italy"]})
+
+    """
+
+    all_groups = ["prior", "posterior"]
+
+    if kind == "observed":
+        all_groups = ["observed_data", "prior_predictive", "posterior_predictive"]
+
+    if coords is None:
+        coords = {}
+
+    if prior_kwargs is None:
+        prior_kwargs = {}
+
+    if posterior_kwargs is None:
+        posterior_kwargs = {}
+
+    if observed_kwargs is None:
+        observed_kwargs = {}
+
+    if backend_kwargs is None:
+        backend_kwargs = {}
+
+    datasets = []
+    groups = []
+    for group in all_groups:
+        try:
+            datasets.append(getattr(data, group))
+            groups.append(group)
+        except:
+            pass
+
+    if var_names is None:
+        var_names = list(datasets[0].data_vars)
+
+    if isinstance(var_names, str):
+        var_names = [var_names]
+
+    if isinstance(var_names[0], str):
+        var_names = [var_names for _ in datasets]
+
+    var_names = [_var_names(vars, dataset) for vars, dataset in zip(var_names, datasets)]
+
+    if transform is not None:
+        datasets = [transform(dataset) for dataset in datasets]
+
+    datasets = get_coords(datasets, coords)
+    pp_plotters = [
+        list(xarray_var_iter(data, var_names=var, combined=True))
+        for data, var in zip(datasets, var_names)
+    ]
+
+    nvars = len(pp_plotters[0])
+    ngroups = len(groups)
+
+    (figsize, ax_labelsize, _, xt_labelsize, linewidth, markersize) = _scale_fig_size(
+        figsize, textsize, 2 * nvars, ngroups
+    )
+
+    posterior_kwargs.setdefault("plot_kwargs", dict())
+    posterior_kwargs["plot_kwargs"].setdefault("color", "C0")
```

Comment:
I know this is currently matplotlib only, but now that conversion_to_hex has been merged, this should be updated to:

```
posterior_kwargs["plot_kwargs"].setdefault("color", "C0")
posterior_kwargs["plot_kwargs"]["color"] = vectorized_to_hex(posterior_kwargs["plot_kwargs"]["color"])
```

to ease the implementation of the bokeh version using the same colors as mpl one

Reply:
In fact, maybe:

```
posterior_kwargs["plot_kwargs"]["color"] = vectorized_to_hex(posterior_kwargs["plot_kwargs"].get("color", "C0"))
```

is more pythonic? I _think_ it does the same. Also, shouldn't `posterior_kwargs["plot_kwargs"]` be dealiased before setting the defaults?

Reply:
Okay, will do. I also thought earlier about dealiasing, but procrastinated it for the end. Will add now.

Reply:
> I think it does the same. Also, shouldn't posterior_kwargs["plot_kwargs"] be dealiased before setting the defaults?

The dealiasing has been handled in plot_kde itself. I don't think we need to dealiase here.

Reply:
If the user passes `c` as part of `plot_kwargs` and we then do the `setdefault` with `color`, what does dealiasing do? Keeps `c` (should be as it is the user introduced one), keeps `color` or crashes?